fvtest/porttest/omrstrTest.cpp

```diff
@@ -1479,39 +1478,47 @@ TEST(PortStrTest, str_WinacpToMutf8)
  *         TEST_FAIL on error
  */
 static int
-printToBuffer(struct OMRPortLibrary *portLibrary, BOOLEAN precision, char *buffer, size_t bufferLength, const char *expectedOutput, const char *format, ...)
+printToBuffer(struct OMRPortLibrary *portLibrary, BOOLEAN precision, char *buffer, size_t bufferLength, const char *expectedOutput, int expectedCount, const char *format, ...)
 {
 	OMRPORT_ACCESS_FROM_OMRPORT(portLibrary);
+	static const char NULLCHARSTRING[] = "[null]";
 	int stringLength = 0;
+	int expectedLength = (NULL == expectedOutput) ? 0 : strlen(expectedOutput);
+	int actualCount = 0;
+	const char *nonNullExpectedOutput = (NULL == expectedOutput) ? NULLCHARSTRING : expectedOutput;
+	const char *nonNullBuffer = (NULL == buffer) ? NULLCHARSTRING : buffer;
 	va_list args;
 
 	va_start(args, format);
-	stringLength = atoe_vsnprintf(buffer, bufferLength, format, args);
+	actualCount = atoe_vsnprintf(buffer, bufferLength, format, args);
+	stringLength = (NULL == buffer) ? 0 : strlen(buffer);
 	va_end(args);
 
-	if (stringLength >= 0) {
+	if (actualCount >= 0) {
 		if (precision) {
-			portTestEnv->log("\n\tFinish Testing: String in buffer is: \"%s\", length = %d\n", buffer, stringLength);
+			portTestEnv->log("\n\tFinish Testing: String in buffer is: \"%s\", length = %d\n", nonNullBuffer, stringLength);
 		} else {
-			portTestEnv->log("\n\tFinish Testing without precision specifier: String in buffer is: \"%s\", length = %d\n", buffer, stringLength);
+			portTestEnv->log("\n\tFinish Testing without precision specifier: String in buffer is: \"%s\", length = %d\n", nonNullBuffer, stringLength);
 		}
 
-		if ((stringLength == strlen(expectedOutput))
-			&& (0 == memcmp(buffer, expectedOutput, stringLength))
+		if ((stringLength == expectedLength)
+			&& (actualCount == expectedCount)
+			&& ((NULL == buffer) || (0 == memcmp(buffer, expectedOutput, stringLength + 1)))
 		) {
 			portTestEnv->log("\n\tComparing against the expected output: PASSED.\n");
 			return TEST_PASS;
 		} else {
-			portTestEnv->log(LEVEL_ERROR, "\n\tComparing against the expected output: FAILED. Expected string: \"%s\", length = %d\n", expectedOutput, strlen(expectedOutput));
+			portTestEnv->log(LEVEL_ERROR, "\n\tComparing against the expected output: FAILED. Expected string: \"%s\", length = %d\n", nonNullExpectedOutput, expectedLength);
 			return TEST_FAIL;
 		}
 	} else {
-		portTestEnv->log(LEVEL_ERROR, "\n\tComparing against the expected output: FAILED. stringLength < 0, Expected string: \"%s\", length = %d\n", expectedOutput, strlen(expectedOutput));
+		portTestEnv->log(LEVEL_ERROR, "\n\tComparing against the expected output: FAILED. stringLength < 0, Expected string: \"%s\", length = %d\n", nonNullExpectedOutput, expectedLength);
 	}
 }
 
 /**
  * Verify that null/non-null terminated strings are properly handled in fstring().
```

Comment:
I'm not sure what the reference to `fstring()` is here. Could you correct the description, please?

Reply:
It is an internal method in the etoa_utils.c file. I remove fstring() from this comment since the exact details of how that service works is not relevant here.